### ISSUE TITLE
InfoRequest#law_used_human doesn't return locale-specific translation

### ISSUE THREAD

**Issue by garethrees** [mysociety/belgium-theme#45]
> In this page https://transparencia.be/nl_BE/request/liste_des_asbl_subsidiees_par_le_19 these is a sentence "AZOUZI Mohamed heeft deze vraag van Liberté d'accès à l'information aan Ville de Bruxelles" at the top.
>
> The words "Liberté d'accès à l'information" are the equivalent of {{law_used_full}} in Transifex and must be translated in Dutch by "Openbaarheid van bestuur".
> 
> I don't think that I can do it by myself via Transifex?
>
> – https://groups.google.com/a/mysociety.org/forum/#!topic/alaveteli/DCsVfjndYt8

So here's what I can establish:

- I'm viewing a request in `nl_BE`. I'd expect to see Dutch, but I don't.
- The app is currently deployed at `0.27.1.3`, which [has the correct Dutch translation](https://github.com/mysociety/alaveteli/blob/0.27.1.3/locale/nl_BE/app.po#L1040-L1041).

Here's what happens in the console:

```ruby
I18n.with_locale(:fr) { InfoRequest.find(53).law_used_human(:full) }
# => "Liberté d'accès à l'information"
I18n.with_locale(:nl_BE) { InfoRequest.find(53).law_used_human(:full) }
# => "Liberté d'accès à l'information"
```

I wonder whether the requets's `User` is having an impact somehow:

```ruby
InfoRequest.find(53).user.locale
# => "fr_BE"
```

**Comment by lizconlan** [comment]
missing model_patch override - see AskTheEU https://github.com/mysociety/asktheeu-theme/blob/master/lib/model_patches.rb

**Comment by garethrees** [comment]
> missing model_patch override?

My first thought was that we shouldn't need that here, as we haven't changed the translation strings themselves?

**Comment by lizconlan** [comment]
ugh, appears to be freezing with the default locale translations rather than the intended behaviour 😢

**Comment by garethrees** [comment]
Maybe try a spec like this and remove `#freeze` to see if it passes?

```ruby
request = FactoryGirl.create(:info_request)
en = I18n.with_locale(:en) { request.law_used_human(:full) }
es = I18n.with_locale(:es) { InfoRequest.find(53).law_used_human(:full) }
expect(en).to eq('…')
expect(es).to eq('…')
```

**Comment by lizconlan** [comment]
@garethrees there's no `.freeze` in the model code, it's just not re-evaluating it once it's been set because we've told the model it's a constant (maybe "caching" would have been a better word but neither's quire right)

**Comment by garethrees** [comment]
Ah, so there isn't! Maybe try converting `LAW_USED_READABLE_DATA` to a method as a tester

**Comment by lizconlan** [comment]
yep, that works!

**Comment by lizconlan** [comment]
Code fixed in release 0.28.0.6

Fix for Belgian site relies on #48

**Comment by garethrees** [comment]
Closing in favour of the upgrade ticket https://github.com/mysociety/belgium-theme/issues/48
